pandas_dev_flaker/_plugins_tree/disallow_argument_types.py

```diff
@@ -0,0 +1,45 @@
+import ast
+from typing import Iterator, Tuple
+
+from pandas_dev_flaker._data_tree import State, register
+
+MSG = "PDF026 found union between Series and AnyArrayLike in type hint"
+
+
+@register(ast.FunctionDef)
+def visit_arg(
+    state: State,
+    node: ast.FunctionDef,
+    parent: ast.AST,
+) -> Iterator[Tuple[int, int, str]]:
+    arguments = node.args.args
+    for arg in arguments:
+        if isinstance(arg.annotation, ast.BinOp) and parse_annotation(
+            arg.annotation,
+        ):
+            yield arg.lineno, arg.col_offset, MSG
+
+
+def parse_annotation(node: ast.BinOp) -> bool:
```

Comment:
also, this probably needs a better name, e.g. `constains_series_and_arraylike`

Reply:
do you think i should change the function name from `visit_args` to `visit_FunctionDef`? seems more appropriate and consistent

Reply:
yeah sounds good